awx/main/tasks.py

```diff
@@ -678,11 +678,12 @@ def should_use_proot(self, instance, **kwargs):
         return False
 
     def build_inventory(self, instance, **kwargs):
-        path = os.path.join(kwargs['private_data_dir'], 'inventory')
-        with open(path, 'w') as f:
-            json_data = json.dumps(instance.inventory.get_script_data(hostvars=True))
-            f.write('#! /usr/bin/env python\n# -*- coding: utf-8 -*-\nprint %r\n' % json_data)
-            os.chmod(path, stat.S_IRUSR | stat.S_IXUSR)
+        json_data = json.dumps(instance.inventory.get_script_data(hostvars=True))
+        handle, path = tempfile.mkstemp(dir=kwargs.get('private_data_dir', None))
+        f = os.fdopen(handle, 'w')
+        f.write('#! /usr/bin/env python\n# -*- coding: utf-8 -*-\nprint %r\n' % json_data)
+        f.close()
+        os.chmod(path, stat.S_IRUSR | stat.S_IXUSR | stat.S_IWUSR)
```

Comment:
https://docs.python.org/2/library/tempfile.html#tempfile.mkstemp

> mkstemp() returns a tuple containing an OS-level handle to an open file (as would be returned by os.open()) and the absolute pathname of that file, in that order.

https://docs.python.org/2/library/os.html#os.fdopen

> Return an open file object connected to the file descriptor fd. 

Just trying to paint an accurate picture of what this is doing. `handle` is a handle to an open file, but its interpretation when passing it to `fdopen` is that it is a file descriptor.

Reply:
`as would be returned by os.open()` but not implying that it is os.open() that is returning the file.